packages/devtools_app/lib/src/shared/globals.dart

```diff
@@ -15,6 +17,22 @@ import 'framework_controller.dart';
 import 'preferences.dart';
 import 'survey.dart';
 
+/// Whether this DevTools build is external.
+bool get isExternalBuild => _isExternalBuild;
+bool _isExternalBuild = false;
+
+/// Flag the build as external.
+set isExternalBuild(bool value) {
+  if (kReleaseMode && _isExternalBuild && !value) {
+    throw StateError(
+      'isExternalBuild should not switch from true to false in release mode.',
+    );
+  }
+  _isExternalBuild = value;
+}
+
+bool isTestEnvironment = true;
```

Comment:
this does not seem right to have a variable isTestEnvironment that is true outside of tests.

Reply:
we should get rid of this and just set the `FeatureFlags.myFlag` value to what we want from the tests. As part of our common setup methods we can add a method `turnOffAllFeatureFlags` or something like that.

I'm trying to think if there is some way we can detect whether we are in a test or not, but I can't think of anything that is more lightweight than my suggestion currently.

Reply:
Yes, it tests all features should be off by default, except cases when we want to test the feature under construction. 

The question is - how to make all features off in tests, if we want them on when the app starts in debug mode.
May be the application should check it and, if it is not release mode, turn them on?

Reply:
> The question is - how to make all features off in tests

For this, my suggestion is to add a testing helper method `turnOffAllFeatureFlags`. It is expected to need to do some setup steps for tests. For example, in several tests we have to set globals like `IdeTheme` that are expected to be non-null. Setting all the feature flags to false for tests (other than when we want to run tests with a flag on) will allow us to keep the value of `kEnableBeta = !isExternalBuild && !kReleaseMode` for the default (non-test) case.